core/include/detray/propagator/rk_stepper.ipp

```diff
@@ -702,50 +702,41 @@ DETRAY_HOST_DEVICE bool detray::rk_stepper<
         const vector3 err_vec =
             one_sixth * h2 *
             (sd.dtds[0u] - sd.dtds[1u] - sd.dtds[2u] + sd.dtds[3u]);
-        error_estimate =
-            math::max(getter::norm(err_vec), static_cast<scalar_type>(1e-20));
+        error_estimate = getter::norm(err_vec);
 
-        return (error_estimate <= cfg.rk_error_tol);
+        return error_estimate;
     };
 
-    // Initial step size estimate
-    stepping.set_step_size(navigation());
+    // std::size_t n_step_trials{0u};
 
-    scalar_type step_size_scaling{1.f};
-    std::size_t n_step_trials{0u};
+    scalar error{1e20f};
 
-    // Adjust initial step size to integration error
-    while (!try_rk4(stepping._step_size)) {
+    // The track entering in the new volume initialize its step size as the
+    // estimated path length to its next expected boundary. We need to reduce it
+    // down to make error small enough
+    if (stepping._initialized == true) {
```

Comment:
Sorry - not for every volume but for every `init()` call.
The reason is that the step size can be very small when it tries reache the surface
It will take many steps to recover to the normal step size -- that's why.

Ideally we need to initialized for every surface intersection not `init()` which is also called for `fair_trust` but I don't find a nice way to implement this

Reply:
That the stepper will take more steps after this change

Reply:
The step size is set to navigation() in the init call. So it is not different from before I guess

Reply:
Ah init() is not called for every surface intersection. Where should I edit in the navigator to release the step for every intersection?

Reply:
nvm I edited the `update_navigation_status` function